packages/react/src/analytics/integrations/GA4/GA4.ts

```diff
@@ -148,9 +150,36 @@ class GA4 extends integrations.Integration<GA4IntegrationOptions> {
       true,
     );
 
+    const dataLayer = get(
+      this.options,
+      OPTION_DATA_LAYER_NAME,
+      DEFAULT_DATA_LAYER_NAME,
+    );
+
+    googleConsentMode.addIntegration(this.loadData.consent, dataLayer);
```

Comment:
I do not understand why this function is called `addIntegration` since we are adding `dataLayer`s here. Also, I think the order of the arguments should be reversed, i.e., dataLayer should be first.